docs: Card a11y docs (#4728)
kiwicom/orbit: docs/src/documentation/03-components/02-structure/card/03-accessibility.mdx

| ------- | :----- | :------------------------------------------------------------------------------------------------------------- |
| titleAs | `enum` | Controls the semantic heading level of the CardSection title, allowing proper document hierarchy within Cards. |

### Keyboard navigation

#### Card component

Cards themselves are not focusable elements, but they may contain interactive elements like buttons, links, or form controls that are part of the card's content. These interactive elements follow standard keyboard navigation:

- Interactive elements within cards can be accessed using the `Tab` key
- When a Card contains the close button, it can be activated using `Enter` or `Space`

#### CardSection component

CardSection components have specific keyboard behaviors when they're expandable:

- Expandable sections can be toggled with `Enter` or `Space` when focused
- Focus is managed automatically when expanding/collapsing sections
- When multiple expandable sections exist in a Card, users can navigate between them using `Tab`

### Screen reader experience

#### Card component

Screen readers will announce:

> - The Card title as specified by the `titleAs` prop (when provided)

Reviewer: This is not true

- The close button with the custom label specified in `labelClose` prop

> - The section identifier provided in `dataA11ySection` (when provided)

Reviewer: Hmm is this true? Isn't this just used for the SkipNavigation?

#### CardSection component

Screen readers will announce:

> - The CardSection title as specified by the `titleAs` prop

Reviewer: Same as above

- The expanded/collapsed state for expandable sections
- The content within the section when expanded

### Important restrictions

> - **CardSection with actions and expandability**: When using the `actions` prop with an expandable CardSection (`expandable={true}` or with `onClick` handler), prop `actions` should be undefined or passed a non-interactive element. This prevents focus management issues and keyboard navigation confusion. Interactive elements in this context can create conflicts with the expansion functionality, leading to poor accessibility.

Reviewer: > When using the `actions` prop with an expandable CardSection (`expandable={true}` or with `onClick` handler), prop `actions` should be undefined or passed a non-interactive element.

I find this sentence quite confusing. How can I be using it but set it to undefined?

Example of what to avoid:

```jsx
<CardSection
  expandable
  title="Flight details"
  actions={<Button>Edit</Button>} // Avoid interactive elements here
>
  {/* content */}
</CardSection>
```

> Recommended approach:

Reviewer: We can also say that a Button can be used in actions, but rendered as a div, with non-interactive behavior, since it is applied automatically by the `expandable` prop and its associated logic

```jsx
<CardSection
  expandable
  title="Flight details"
  actions={<Badge type="info">Economy</Badge>} // Non-interactive element is acceptable
>
  {/* content */}
</CardSection>
```

### Best practices

#### Card best practices

- Use semantic heading levels with the `titleAs` prop to maintain proper document structure
- Provide a descriptive `labelClose` that clearly explains the action (e.g., "Close flight details")

> - When creating complex card layouts and using SkipNavigation component on the page, you can use the `dataA11ySection` prop connect the Card with this component

Reviewer: ```suggestion
- When creating complex page layouts and using SkipNavigation component on the page, you can use the `dataA11ySection` prop connect the Card with this component
```

#### CardSection best practices

- Avoid placing interactive elements in the `actions` prop of expandable CardSection components
- Maintain logical heading hierarchy between Card titles and CardSection titles

### Examples with accessibility features

#### Basic Card with accessibility props

```jsx
<Card
  title="Flight details"
  titleAs="h3"
  dataA11ySection="flight-summary"
  labelClose="Close flight details"
  onClose={() => {}}
>
  <CardSection>
    <Stack direction="column" spacing="small">
      <Text>Prague to Barcelona</Text>
      <Text type="secondary">June 21, 2023</Text>
    </Stack>
  </CardSection>
</Card>
```

> Screen readers will announce this as a section identified as `flight-summary` with a heading level 3 of "Flight details", followed by the content, with a close button labeled "Close flight details".

Reviewer: Do screen readers announce heading levels? And what does "announce this as a section identified as `flight-summary`" mean exactly?

Author: yes, VO announces the level of the heading. the `flight-summary` in dataA11ySection was polished.
<img width="802" alt="image" src="https://github.com/user-attachments/assets/930791b1-3d63-4f49-a41b-559fdc28cb41" />

- Interactive elements within cards can be accessed using the `Tab` key
- When a Card contains the close button, it can be activated using `Enter` or `Space`

#### CardSection component

CardSection components have specific keyboard behaviors when they're expandable:

- Expandable sections can be toggled with `Enter` or `Space` when focused
- Focus is managed automatically when expanding/collapsing sections
- When multiple expandable sections exist in a Card, users can navigate between them using `Tab`

### Screen reader experience

#### Card component

Screen readers will announce:

- Card title level as specified by the `titleAs` prop (when provided anything other than `div`)
- The Card title as specified by the `title` prop (when provided)
- The close button with the custom label specified in `labelClose` prop

#### CardSection component

Screen readers will announce:

- The CardSection title as specified by the `title` prop
- The content within the section when expanded
- The expanded/collapsed state for expandable sections

### Important restrictions

> - **CardSection with actions and expandability**: When using the `actions` prop with an expandable CardSection (`expandable={true}` or with `onClick` handler), avoid using the actions prop. If you must use it, ensure it only contains non-interactive elements. This prevents focus management issues and keyboard navigation confusion. Interactive elements in this context can create conflicts with the expansion functionality, leading to poor accessibility.

Reviewer: ```suggestion
- **CardSection with actions and expandability**: When using a clickable or expandable CardSection (`expandable={true}` or with `onClick` handler), avoid using the `actions` prop. If you must use it, ensure it only contains non-interactive elements. This prevents focus management issues and keyboard navigation confusion. Interactive elements in this context can create conflicts with the expanding functionality, leading to poor accessibility and usability.
```

Example of what to avoid:

```jsx
<CardSection
  expandable
  title="Flight details"
  actions={<Button>Edit</Button>} // Avoid interactive elements here
>
  {/* content */}
</CardSection>
```

Recommended approach:

```jsx
<CardSection
  expandable
  title="Flight details"
  actions={<Badge type="info">Economy</Badge>} // Non-interactive element is acceptable
>
  {/* content */}
</CardSection>
```

> Also, Button component can be passed to the `actions` prop, but it should be non-interactive (ie. setting `asComponent="div"`).

Reviewer: ```suggestion
Also, Button component can be passed to the `actions` prop, but it should be non-interactive (ie. setting `asComponent="div"` and not defining `onClick`).
```
